From 7162aca8db617a5c6004a80571bc2a06588e7afb Mon Sep 17 00:00:00 2001
From: Andrew Kelley <andrew@ziglang.org>
Date: Mon, 13 Oct 2025 22:10:29 -0700
Subject: [PATCH 024/133] coff linker: don't check the time
X-Developer-Signature: v=1; a=openpgp-sha256; l=1326; i=xtex@astrafall.org;
 h=from:subject; bh=GRCjrcikEuQOPMRSUDowKgAvkMEI28Q1Ztl2rBElFXg=;
 b=owGbwMvMwCW2U4Ij7wZL9ETG02pJDJn7cn8z7Zl19Z26lO5fRdk00V2rJ3w8+HVhzeuLffymp
 5pnrdLJ6pjIwiDGxWAppshSZNjgzaqTzi+6rFwWZg4rE8gQaZEGBgYGBhYGvtzEvFIjHSM9U21D
 PUNDHSMdIwYuTgGY6nmujAyztE3am//oZrfP2ca98238jitdm47Z31n5+diOri75D1zSDH/l31q
 v3ciTs7fx4WTlw86rK7tXNi8WPLf/ZtQvbaP0rds5AA==
X-Developer-Key: i=xtex@astrafall.org; a=openpgp;
 fpr=7231804B052C670F15A6771DB918086ED8045B91

compiler toolchains have no business knowing what time it is

(cherry picked from commit 79a59c516531650664511ac539ee6d7c6402169f)
Change-Id: If9c7b187f46191367063cbbbb71254eebea0443f
Signed-off-by: xtex <xtexchooser@duck.com>
---
 src/link/Coff.zig | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/link/Coff.zig b/src/link/Coff.zig
index 62430f6c08e6..e1d91a6c3aec 100644
--- a/src/link/Coff.zig
+++ b/src/link/Coff.zig
@@ -2200,12 +2200,12 @@ fn writeHeader(coff: *Coff) !void {
         flags.DLL = 1;
     }
 
-    const timestamp = if (coff.repro) 0 else std.time.timestamp();
+    const timestamp: u32 = 0;
     const size_of_optional_header = @as(u16, @intCast(coff.getOptionalHeaderSize() + coff.getDataDirectoryHeadersSize()));
     var coff_header = coff_util.CoffHeader{
         .machine = target.toCoffMachine(),
         .number_of_sections = @as(u16, @intCast(coff.sections.slice().len)), // TODO what if we prune a section
-        .time_date_stamp = @as(u32, @truncate(@as(u64, @bitCast(timestamp)))),
+        .time_date_stamp = @as(u32, @truncate(@as(u32, @bitCast(timestamp)))),
         .pointer_to_symbol_table = coff.strtab_offset orelse 0,
         .number_of_symbols = 0,
         .size_of_optional_header = size_of_optional_header,
-- 
2.52.0

